controllers/configurationpolicy_utils.go

```diff
@@ -249,8 +249,8 @@ func checkListFieldsWithSort(mergedObj []map[string]interface{}, oldObj []map[st
 				}
 			case map[string]interface{}:
 				// if a map in the list contains another map, check fields for equality
-				if !checkFieldsWithSort(mVal, oldItem[i].(map[string]interface{})) {
-					return false
+				if oVal, ok := oldItem[i].(map[string]interface{}); ok {
+					return len(mVal) == len(oVal) && checkFieldsWithSort(mVal, oVal)
```

Comment:
Would it need to return `false` after this since `ok` would have failed, similar to above?

Reply:
Probably. Good catch.

I started to get distracted making all of these kinds of things more consistent... but I think that it needs more attention than I can give it right now. It is probably due for some refactoring to reduce some repeated logic like length checks before and in `checkListsMatch`. So, maybe that's a future item.